### PR TITLE
feat: add ubuntu minimal gen1/gen2 VHD build support

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**
This PR adds Ubuntu Minimal Gen1/Gen2 VHD build support to our pipelines. Green build https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=75675653&view=results

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cis.sh [hunk 1]
```diff
@@ -61,7 +61,9 @@ assignFilePermissions() {
         chmod 0600 /etc/crontab || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
     fi
     for filepath in /etc/cron.hourly /etc/cron.daily /etc/cron.weekly /etc/cron.monthly /etc/cron.d; do
-        chmod 0600 $filepath || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/cis.sh:64]: hah I remember having to do something similar in my old PR for minimal

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 2]
```diff
@@ -706,4 +714,56 @@ stages:
...
+      steps:
+        - bash: |
+            echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
+            echo '##vso[task.setvariable variable=OS_SKU]Ubuntu'
+            echo '##vso[task.setvariable variable=OS_VERSION]22.04'
+            echo '##vso[task.setvariable variable=IMG_PUBLISHER]Canonical'
+            echo '##vso[task.setvariable variable=IMG_OFFER]0001-com-ubuntu-minimal-jammy-aks-daily-preview'
+            echo '##vso[task.setvariable variable=IMG_SKU]minimal-aks-22_04-daily-lts'
+            echo '##vso[task.setvariable variable=IMG_VERSION]latest'
+            echo '##vso[task.setvariable variable=HYPERV_GENERATION]V1'
```
Comment (cameronmeissner) [inline .pipelines/.vsts-vhd-builder-release.yaml:733]: shouldn't this be `V2`?

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 3]
```diff
@@ -706,4 +714,56 @@ stages:
...
+      steps:
+        - bash: |
+            echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
+            echo '##vso[task.setvariable variable=OS_SKU]Ubuntu'
+            echo '##vso[task.setvariable variable=OS_VERSION]22.04'
+            echo '##vso[task.setvariable variable=IMG_PUBLISHER]Canonical'
+            echo '##vso[task.setvariable variable=IMG_OFFER]0001-com-ubuntu-minimal-jammy-aks-daily-preview'
+            echo '##vso[task.setvariable variable=IMG_SKU]minimal-aks-22_04-daily-lts-gen2'
+            echo '##vso[task.setvariable variable=IMG_VERSION]latest'
+            echo '##vso[task.setvariable variable=HYPERV_GENERATION]V1'
```
Comment (cameronmeissner) [inline .pipelines/.vsts-vhd-builder-release.yaml:759]: shouldn't this be `V2`?

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 4]
```diff
@@ -706,4 +714,56 @@ stages:
...
+            echo '##vso[task.setvariable variable=AZURE_VM_SIZE]Standard_D2s_v3'
+            echo '##vso[task.setvariable variable=FEATURE_FLAGS]None'
+            echo '##vso[task.setvariable variable=CONTAINER_RUNTIME]containerd'
+            echo '##vso[task.setvariable variable=ARCHITECTURE]X86_64'
+            echo '##vso[task.setvariable variable=ENABLE_TRUSTED_LAUNCH]False'
+            echo '##vso[task.setvariable variable=SGX_INSTALL]False'
+          displayName: Setup Build Variables
+        - template: ./templates/.builder-release-template.yaml
+          parameters:
+            artifactName: 2204-Minimal-Gen1-Containerd
```
Comment (alexeldeib) [inline .pipelines/.vsts-vhd-builder-release.yaml:743]: nit: inconsistent casing with all other skus

File: .pipelines/.vsts-vhd-builder-release.yaml (hunk 3 above)
Comment (alexeldeib) [reply .pipelines/.vsts-vhd-builder-release.yaml:759]: yeah -- one of the offers needs `-gen2` and `V2`. you have `V1` duped here.

File: vhdbuilder/packer/install-dependencies.sh [hunk 5]
```diff
@@ -42,7 +42,23 @@ APT::Periodic::Unattended-Upgrade "0";
...
+# If the IMG_SKU does not contain "minimal", installDeps normally
+if [[ "$IMG_SKU" != *"minimal"* ]]; then
+  installDeps
+else
+  # The following packages are required for an Ubuntu Minimal Image to build and successfully run CSE
+  # jq - for manipulation JSON data
+  # iptables - required to run containerd
+  # netcat - network comms with API server
+  # dnsutils - contains nslookup, to query API server DNS
+  required_pkg_list=(jq iptables netcat dnsutils)
```
Comment (alexeldeib) [inline vhdbuilder/packer/install-dependencies.sh:54]: nice

File: .pipelines/.vsts-vhd-builder-release.yaml (hunk 3 above)
Comment (anujmaheshwari1) [reply .pipelines/.vsts-vhd-builder-release.yaml]: changed, thanks! copy paste shenanigans

File: .pipelines/.vsts-vhd-builder-release.yaml (hunk 4 above)
Comment (anujmaheshwari1) [reply .pipelines/.vsts-vhd-builder-release.yaml]: fixed!

**Review by alexeldeib — APPROVED** [review]

**Review by cameronmeissner — APPROVED** [review]

File: .pipelines/templates/.builder-release-template.yaml [hunk 6]
```diff
@@ -40,6 +40,7 @@ steps:
       if [[ ${OS_VERSION} == "V2" && ${OS_SKU} == "CBLMariner" && ${ENABLE_FIPS,,} == "true" ]]; then SKU_NAME="${SKU_NAME}fips"; fi && \
       if [[ "$(FEATURE_FLAGS)" == *"fullgpu"* ]]; then SKU_NAME="${SKU_NAME}gpu"; fi && \
       if [[ "${IMG_SKU}" == "20_04-lts-cvm" ]]; then SKU_NAME="${SKU_NAME}CVM"; fi && \
+      if [[ "${IMG_SKU}" == *"minimal"* ]]; then SKU_NAME="${SKU_NAME}minimal"; fi && \
```
Comment (anujmaheshwari1) [inline .pipelines/templates/.builder-release-template.yaml:43]: @cameronmeissner

**Pull request merged**